components/EditISCNMetadataModal.vue

```diff
@@ -0,0 +1,256 @@
+
+          // Parse sameAs URLs into downloadableUrls
+          const downloadableUrls = metadata.sameAs
+            ? metadata.sameAs.map(parseDownloadableUrl)
+            : [
+                {
+                  url: '',
+                  type: '',
+                  fileName: ''
+                }
+              ]
+
+          iscnData.value = {
+            type: metadata['@type'] || 'Book',
+            title: metadata.name || '',
+            description: metadata.description || '',
+            isbn: metadata.isbn || '',
+            publisher: metadata.publisher || '',
+            publicationDate: metadata.datePublished || '',
+            author: {
+              name: metadata.author?.name || metadata.author || '',
+              description: metadata.author?.description || ''
+            },
+            license: metadata.usageInfo || 'All Rights Reserved',
+            contentFingerprints: record.data.contentFingerprints.map(url => ({
+              url
+            })) || [{ url: '' }],
+            downloadableUrls,
+            customLicense: '',
+            language: metadata.inLanguage || '',
+            bookInfoUrl: metadata.url || '',
+            tags: metadata.keywords
+              ? metadata.keywords.split(',').map(k => k.trim())
+              : [],
+            coverUrl: metadata.thumbnailUrl || ''
+          }
+        }
+      }
+    } catch (error) {
+      console.error('Error fetching ISCN data:', error)
+    } finally {
+      isISCNLoading.value = false
+    }
+  }
+})
+
+function handleClickBack () {
+  emit('update:modelValue', false)
+}
+
+function parseDownloadableUrl (url: string) {
+  try {
+    const urlObj = new URL(url)
+    const fileName = urlObj.searchParams.get('name') || ''
+    const [name, type] = fileName.split('.')
+    return {
+      url: url.split('?')[0],
+      fileName: name,
+      type: type || ''
+    }
+  } catch (e) {
+    return {
+      url,
+      fileName: '',
+      type: ''
+    }
+  }
+}
+
+async function handleSave () {
+  await initIfNecessary()
+  if (!wallet.value || !signer.value) {
+    toast.add({
+      title: 'Please login first',
+      color: 'red'
+    })
+    return
+  }
+  if (!isFormValid.value) {
+    toast.add({
+      title: 'Please fill in all required fields',
+      color: 'red'
+    })
+    return
+  }
+  isSaving.value = true
+  try {
+    await signISCNTx(
+      formatISCNTxPayload(payload.value),
+      signer.value,
+      wallet.value,
+      {
+        iscnId: `${iscnId.value}/${recordVersion.value}`,
+        gas: new BigNumber(ISCN_GAS_FEE)
+          .multipliedBy(UPDATE_ISCN_GAS_MULTIPLIER)
+          .toFixed(0)
+      }
+    )
+    if (props.classId) {
+      classData.value = await nftStore.fetchClassMetadataById(props.classId as string)
+    }
+    toast.add({
+      title: 'ISCN updated successfully',
+      color: 'green'
+    })
+    emit('save', `${iscnId.value}/${recordVersion.value + 1}`)
```

Comment:
Do all listeners of `save` need the `recordVersion` for their usecase?

Reply:
[♻️ Simplify](https://github.com/likecoin/nft-book-press/pull/317/commits/62be52a5d91c0ba3a483436538c466b7d563abb8) @save [by extracting parameters](https://github.com/likecoin/nft-book-press/pull/317/commits/62be52a5d91c0ba3a483436538c466b7d563abb8)